### DISCUSSION TITLE
Changing and Resetting Button Colours

### DISCUSSION THREAD

**Discussion by bensuffolk** [HASwitchPlate/openHASP#376] [Q&A] [answered]
I need to be able to colour a button based on a condition. I know I can do this by sending a command to it e.g, **bg_color=red** or some other colour. But what I don't know how to do is to then revert that colour back to the theme colour.

Is these a reset type command, or a way to read the existing property of an object?

Failing that, does anybody know the colour of a button when using the Material Dark Theme?

**Comment by fvanroie** [comment, marked as answer]
> Is there a reset type command

There isn't:
- You either need to get the original `bg_color` first and store that.
- Or remove and recreate the button without a `bg_color`.

>  or a way to read the existing property of an object?

Send the `p12.bg_color` command _without_ any parameters

**Reply by bensuffolk** [reply]
Perfect, if I can read a buttons colour first that’s all I need to be able to do.
